oak_functions/loader/src/metrics.rs

```diff
@@ -0,0 +1,169 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::logger::Logger;
+use log::Level;
+use rand::{distributions::Open01, thread_rng, Rng};
+use serde::Deserialize;
+use std::{
+    collections::{HashMap, HashSet},
+    sync::{Arc, Mutex},
+};
+
+/// Configuration for differentially-private metrics reporting.
+#[derive(Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct PrivateMetricsConfig {
+    pub epsilon: f64,
+    pub batch_size: usize,
+    pub allowed_labels: Vec<String>,
+}
+
+/// Aggregator for count-based differentially private metrics.
+pub struct PrivateMetricsAggregator {
+    count: usize,
+    epsilon: f64,
+    batch_size: usize,
+    allowed_labels: HashSet<String>,
+    label_count: f64,
+    events: HashMap<String, usize>,
```

Comment:
An alternative would be to just populate the hashmap with one entry per label, initialized to zero; then when incrementing, only use entries that are already present (possibly to zero, but still present). Basically, there is no need for an additional Set. Up to you anyways if you prefer the current approach. Also label_count seems redundant.